src/test/java/wolox/training/UserControllerIntegrationTest.java

```diff
@@ -0,0 +1,68 @@
+package wolox.training;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import wolox.training.controllers.UserController;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.exceptions.UserNotFoundException;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(UserController.class)
+public class UserControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mvc;
+
```

Comment:
Utilice @MockBean para IUserService para evitar integrarse con otras capas de la aplicación.
En cada prueba debemos indicar el comportamiento que queremos darle a cada método del servicio de usuario que este controlador utiliza internamente a través de Mokito.

Reply:
Annotation added to IUserService